src/app/api/admin/[...slug]/route.js

```diff
@@ -2,23 +2,28 @@ import { defaultHandler } from 'ra-data-simple-prisma';
 import { NextResponse } from 'next/server';
 import { prisma } from '@/lib/db';
 import { auth } from '@/lib/auth';
+import { withErrorHandler } from '@/lib/errors/errorHandler';
+import { NotAuthorizedException } from '@/lib/errors/NotAuthorizedException';
+import { MODEL_INCLUDES } from '@/lib/consts';
+import { searchInputFilters } from '@/lib/searchInputFilters';
 
-const handler = auth(async (req) => {
-  if (!req.auth) {
-    return NextResponse.json(
-      { message: 'unauthorized' },
-      {
-        status: 401,
-      },
-    );
-  }
-  try {
+const handler = auth(
+  withErrorHandler(async (req) => {
+    if (!req.auth) throw new NotAuthorizedException();
     const json = await req.json();
-    const result = await defaultHandler(json, prisma);
+    const { resource: modelName } = json;
+    const result = await defaultHandler(json, prisma, {
+      getList: {
+        debug: false,
+        where: searchInputFilters(modelName, json.params?.filter?.q),
+      },
+      getOne: { debug: false, include: MODEL_INCLUDES[modelName] },
```

Comment:
🟠 is that the thing that resolves mtm queries?

Reply:
if it is so, maybe it is also useful for other queries, like `getMany` or something like that?

Reply:
Yes, it is for resolving mtm queries. But, as I understand, we dont have getMany queries at this time. The one who will need to implement this can easily use the same approach

Reply:
makes sense to keep filters `above` specialists and organisations then, but no need to put it in `lib` i guess, its should not ( and cannot ) be reusable for whole project, just for its part

Reply:
@alex-popov-tech as far as it is used only in admin/[...slug] route, i suggest move this function in route.js file of admin/[...slug] rote

Reply:
Agree